src/CLI/Command/DebugExpression.php

```diff
@@ -0,0 +1,78 @@
+<?php
+declare(strict_types=1);
+
+namespace Arkitect\CLI\Command;
+
+use Arkitect\Analyzer\FileParserFactory;
+use Arkitect\ClassSet;
+use Arkitect\CLI\TargetPhpVersion;
+use Arkitect\Rules\Violations;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class DebugExpression extends Command
+{
+    /** @var string|null */
+    public static $defaultName = 'debug:expression';
+
+    /** @var string|null */
+    public static $defaultDescription = <<< 'EOT'
+Check which classes respect an expression
+EOT;
+
+    /** @var string */
+    public static $help = <<< 'EOT'
+Check which classes respect an expression
+EOT;
+
+    protected function configure(): void
+    {
+        $this
+            ->setHelp(self::$help)
+            ->addArgument('expression', InputArgument::REQUIRED)
+            ->addArgument('arguments', InputArgument::IS_ARRAY)
+            ->addOption(
+                'from-dir',
+                'd',
+                InputOption::VALUE_REQUIRED,
+                'The folder in which to search the classes',
+                '.'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        // TODO come gestiamo le diverse versioni di PHP? Con una opzione?
+        $targetPhpVersion = TargetPhpVersion::create(null);
+        $fileParser = FileParserFactory::createFileParser($targetPhpVersion);
+
+        $classSet = ClassSet::fromDir($input->getOption('from-dir'));
+        foreach ($classSet as $file) {
+            $fileParser->parse($file->getContents(), $file->getRelativePathname());
+            $parsedErrors = $fileParser->getParsingErrors();
+
+            foreach ($parsedErrors as $parsedError) {
+                // TODO qua ce ne vogliamo fare qualcosa?
```

Comment:
We can print an error at the end of the process maybe at the moment

Reply:
I print it at the beginning, grouped under a warning